File: app/src/main/java/com/tutv/android/ui/series/SeriesView.java
package com.tutv.android.ui.series;

import com.tutv.android.domain.Episode;
import com.tutv.android.domain.Season;

import java.util.List;

public interface SeriesView {

    void showSeriesName(String seriesName);
    void showSeriesDescription(String seriesDescription);
    void showUserFollows(boolean follows);
    void showFollowerCount(int followers);
    void bindSeasons(final List<Season> seasonList);
    void showSeriesBanner(String bannerUrl);
    void bindSeason(Season season);
    void showSeriesFollowed(boolean followed);
    void showError(String error);

}

File: app/src/main/java/com/tutv/android/ui/series/SeriesView.kt
package com.tutv.android.ui.series

import com.tutv.android.domain.Review
import com.tutv.android.domain.Season

interface SeriesView {
    fun showSeriesName(seriesName: String?)
    fun showSeriesDescription(seriesDescription: String?)
    fun showFollowerCount(followers: Int?)
    fun bindSeasons(seasonList: List<Season>?)
    fun bindReviews(reviewList: List<Review>?)
    fun showSeriesBanner(bannerUrl: String?)
    fun bindSeason(season: Season?)
    fun showSeriesFollowed(followed: Boolean)
    fun showError(error: String?)
}